sources/packages/backend/libs/integrations/src/services/disbursement-value/disbursement-value.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from "@nestjs/common";
+import { IsNull, MoreThan, Not, Repository } from "typeorm";
+import { DisbursementValue } from "@sims/sims-db";
+import { InjectRepository } from "@nestjs/typeorm";
+
+/**
+ * Service layer for Disbursement value.
+ */
+@Injectable()
+export class DisbursementValueService {
+  constructor(
+    @InjectRepository(DisbursementValue)
+    private readonly disbursementValueRepo: Repository<DisbursementValue>,
+  ) {}
+
+  /**
+   * Checks if there is any partial or full award amount, that
+   * was withheld due to a restriction during an e-cert generation.
+   * @param disbursementScheduleId disbursement schedule id.
+   * @returns true if there is any partial or full award amount, that
+   * was withheld due to a restriction.
+   */
+  async hasAnyFullOrPartialAwardWithheldDueToRestriction(
+    disbursementScheduleId: number,
+  ): Promise<boolean> {
+    return !!(await this.disbursementValueRepo.findOne({
```

Comment:
using exists and also const

Reply:
changed to exists, do we need a const with exist? It's a boolean, so we return directly right?